krgpg/krgpg.go

```diff
@@ -0,0 +1,313 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"strings"
+	"time"
+
+	"github.com/urfave/cli"
+
+	"github.com/kryptco/kr"
+	"github.com/kryptco/kr/krdclient"
+)
+
+var stderr *os.File
+
+func setupTTY() {
+	var err error
+	stderr, err = os.OpenFile(os.Getenv("GPG_TTY"), os.O_RDWR, 0)
+	if err != nil {
+		stderr, err = os.OpenFile(os.Getenv("TTY"), os.O_RDWR, 0)
+		if err != nil {
+			stderr = os.Stderr
+		}
+	}
+}
+
+func readLineSplittingFirstToken(reader *bufio.Reader) (firstToken string, rest string, err error) {
+	line, err := reader.ReadString('\n')
+	if err != nil {
+		return
+	}
+	toks := strings.Fields(line)
+	if len(toks) == 0 {
+		err = fmt.Errorf("no tokens")
+		return
+	}
+	firstToken = toks[0]
+	rest = strings.Join(toks[1:], " ")
+	return
+}
+
+func main() {
+	setupTTY()
+	app := cli.NewApp()
+	app.Name = "krgpg"
+	app.Usage = "Sign git commits with your Kryptonite key"
+	app.Flags = []cli.Flag{
+		cli.BoolFlag{
+			Name:  "a",
+			Usage: "Ouput ascii armor",
+		},
+		cli.BoolFlag{
+			Name:  "b,detach-sign",
+			Usage: "Create a detached signature",
+		},
+		cli.BoolFlag{
+			Name:  "s,sign",
+			Usage: "Create a signature",
+		},
+		cli.StringFlag{
+			Name:  "u,local-user",
+			Value: "",
+			Usage: "User ID",
+		},
+		cli.StringFlag{
+			Name:  "status-fd",
+			Value: "",
+			Usage: "status file descriptor",
+		},
+		cli.BoolFlag{
+			Name:  "bsau",
+			Usage: "Git method of passing in detach-sign ascii armor flags",
+		},
+		cli.BoolFlag{
+			Name:  "verify",
+			Usage: "Verify a signature",
+		},
+		cli.StringFlag{
+			Name:  "keyid-format",
+			Usage: "Key ID format",
+		},
+	}
+	app.Action = func(c *cli.Context) error {
+		if c.Bool("bsau") || (c.Bool("s") && c.Bool("b") && c.Bool("a")) {
+			//	TODO: verify userID matches stored kryptonite PGP key
```

Comment:
Delay until later since we always assume the user wants to use the Kryptonite key to sign and we automatically sign new userIDs on the phone. If a user wants to use a local key, they can set `gpg.program` to `gpg` or `gpg2` in that repo's git config.